🚀 1단계 - 용어 사전 만들기

jordy-torvalds opened the pull request.

jordy-torvalds committed:
docs: 용서 사전 정리

hyunssooo commented on README.md:
```diff
@@ -96,8 +96,129 @@ docker compose -p kitchenpos up -d
+| 이름    | name       | 메뉴 그룹의 이름                        |
+| 등록    | create     | 새로운 메뉴 그룹을 추가하는 것                |
+
+### 메뉴 구성 상품 용어 사전
+
+| 한글명      | 영문명          | 설명                       |
+|----------|--------------|--------------------------|
+| 메뉴 구성 상품 | menu product | 메뉴를 구성하는 상품              |
+| 개수       | quantity     | 메뉴 구성 상품가 포함하는 특정 상품의 개수 |
+
+### 메뉴 용어 사전
+
```
`메뉴 구성 상품`과 `메뉴`가 나눠진 이유가 있을까요? 
`메뉴` 하위에 `메뉴 구성 상품`이 포함되는 것 같아서요!

jordy-torvalds replied:
넵 그게 더 나을 수 있을 거 같네요!

hyunssooo commented on README.md:
```diff
@@ -96,8 +96,129 @@ docker compose -p kitchenpos up -d
+
+| 한글명      | 영문명          | 설명                                  |
+|----------|--------------|-------------------------------------|
+| 메뉴 그룹    | menu group   | 유사한 특징을 가져서 동일하게 분류될 수 있는 메뉴의 묶음    |
+| 메뉴 구성 상품 | menu product | 메뉴를 구성하는 상품                         |
+| 메뉴       | menu         | 키친포스에서 판매되는 메뉴                      |
```
문서 상위에 작성해 주셨는데 또 메뉴에 포함되는 이유가 있을까요? 
이보다 메뉴의 설명에 작성하는 것은 어떨까요? 
`하나 이상의 메뉴 항목으로 구성되고 하나의 메뉴 그룹에 속한다.`

jordy-torvalds replied:
네넵 함께 있는게 더 나을거 같네요!

jordy-torvalds replied:
아 메뉴 하나에 다 담으려고 하니 이름이란 단어에 대해 설명을 작성할 때 아래와 같은 고민이 되어서 이럴거면 나눠 쓰자 생각했던거 같아요 !
- 메뉴의 이름?
- 메뉴 그룹의 이름?
- 그냥 이름으로 써야 하나?

hyunssooo commented on README.md:
```diff
@@ -96,8 +96,129 @@ docker compose -p kitchenpos up -d
+| 배달       | delivery         | 주문 유형의 하나로, 주문된 메뉴를 배달주소로 배달 하는 것        |
+| 배달 주소    | delivery address | 고객에 배달 주문에 대해 배달을 요청한 주소                 |
+| 주문 상태    | order status     | 주문이 처리되는 과정에서의 상태를 나타내는 것                |
+
+#### 주문 유형 별 주문 상태 용어 사전
+
+주문 유형에 따라 주문 상태가 가지는 의미가 달라 별도의 용어 사전으로 정리합니다.
```
주문 유형별로 나눠주셨네요! 👍 
어떤 시각으로 나눠주셨을까요? `매장 식사`와 `포장` 의 경우 거의 유사한데 합쳐도 괜찮지 않을까요?
`배달`까지 본다면 중복되는 부분도 많은 것 같아서요! 원식님 생각이 궁금해요. 🙂

jordy-torvalds replied:
하나로 합쳤을 때 조금씩 차이나는 부분을 다 작성해주기 보단 나눠서 보는게 오히려 명확할 수 있을거 같아서 나눴었어요!
합쳐서 보는 것도 한 번 고민해보겠습니다~!

hyunssooo commented on README.md:
```diff
@@ -96,8 +96,129 @@ docker compose -p kitchenpos up -d
+| 한글명   | 영문명           | 설명                 |
+|-------|---------------|--------------------|
+| 비속어   | profanity     | 상스럽고 거친 용어         |
+| 키친라이더 | kitchen rider | 키친 포스의 배달을 담당하는 업체 |
+| 라이더   | rider         | 주문을 배달하는 대상        |
```
배달에서만 필요한 용어들 같은데요. 기타로 빠진 이유가 있을까요? 🤔

jordy-torvalds replied:
다시 보니 배달에 있는게 더 나을거 같네요!!

hyunssooo reviewed (changes requested):
안녕하세요. 원식님
이번 미션 함께 진행할 정현수입니다. 🙂
전체적으로 용어 사전 정리 잘 해주셨어요! 
몇 가지 코멘트 남겼는데요. 확인 부탁드려요. 
혹시 페어로 진행하셨을까요?

jordy-torvalds commented:
저랑 페어로 진행하시는 분이 개인사정상 함께 진행하기 어려우실거 같다고 하셔서 별도로 진행하려 합니다!

jordy-torvalds commented:
![Uploading image.png…]()

jordy-torvalds commented:
<img width="466" alt="image" src="https://github.com/next-step/ddd-strategic-design/assets/58139899/95f3cf00-1896-4495-9dbb-6e828ddcbf44">

jordy-torvalds committed:
docs: 피드백 반영

- 메뉴 구성 상품을 메뉴 하위로 이전
- 주문 유형 별로 상태를 정리하는 것이 아닌 하나의 테이블로 정리
- 기타에 작성된 키친라이더와 라이더 단어를 주문으로 이전

hyunssooo commented on README.md:
```diff
@@ -96,8 +96,99 @@ docker compose -p kitchenpos up -d
+### 메뉴 용어 사전
+
+| 한글명      | 영문명          | 설명                                                   |
+|----------|--------------|------------------------------------------------------|
+| 메뉴       | menu         | 키친포스에서 판매되는 메뉴로 하나의 메뉴 그룹과 한 개 이상의 메뉴 구성 상품으로 구성됩니다. |
```
이 부분은 수정되면 좋을 것 같아요. 
`하나의 메뉴 그룹에 속하고`

hyunssooo commented on README.md:
```diff
@@ -96,8 +96,99 @@ docker compose -p kitchenpos up -d
+| 등록       | create       | 새로운 데이터를 추가하는 것.                                     |
+| 가격 변경    | change price | 메뉴가 고객에게 판매되는 가격을 변경하는 것.                            |
+
+#### 메뉴 그룹 용어 사전
+
+| 한글명   | 영문명        | 설명                               |
+|-------|------------|----------------------------------|
+| 메뉴 그룹 | menu group | 유사한 특징을 가져서 동일하게 분류될 수 있는 메뉴의 묶음 |
+| 이름    | name       | 메뉴 그룹의 이름                        |
+| 등록    | create     | 새로운 메뉴 그룹을 추가하는 것                |
```
메뉴 그룹은 메뉴의 하위로 분류되는 걸까요? 하위로 분류된다면 어떤 기준으로 나눠주셨을까요? 
컨텍스트별로 작성한다고 해도 메뉴 그룹은 메뉴의 하위 개념은 아닌 것 같아서요.

jordy-torvalds replied:
다시 생각해보니 생성과 변경 시점이 다르므로 나눠지는게 맞겠네요!

hyunssooo commented on README.md:
```diff
@@ -96,8 +96,99 @@ docker compose -p kitchenpos up -d
+| 메뉴 구성 상품 | menu product | 메뉴를 구성하는 상품              |
+| 개수       | quantity     | 메뉴 구성 상품가 포함하는 특정 상품의 개수 |
+
+### 주문 용어 사전
+
+| 한글명      | 영문명              | 설명                                                   |
+|----------|------------------|------------------------------------------------------|
+| 주문       | order            | 노출 중인 메뉴에 대해 판매가 발생한 것. 한 개 이상의 주문 품목으로 구성됩니다.       |
+| 주문 유형    | order type       | 고객에 주문된 메뉴를 얻는 방법                                    |
+| 주문 발생 시점 | order date time  | 주문이 발생한 시각                                           |
+| 매장식사     | eat in           | 주문 유형의 하나로, 주문된 메뉴를 매장의 주문 테이블에서 먹고 가는 것             |
+| 포장       | takeout          | 주문 유형의 하나로, 주문된 메뉴를 포장해서 가져가는 것                      |
+| 배달       | delivery         | 주문 유형의 하나로, 주문된 메뉴를 배달주소로 배달 하는 것                    |
```
https://github.com/next-step/ddd-strategic-design/pull/267#discussion_r1313902172
이 코멘트가 조금 오해가 있었던 것 같은데요. 
어떤 기준으로 나눴는지 기준이 궁금해서 여쭤본 거였습니다. 🥲 
이전에 작성해 주신 용어 사전도 좋았습니다. 

합치기로 결정하셨으면 주문 유형에 따라서 주문 상태를 작성해 주셔도 좋을 것 같습니다. 

배달 주문: `대기` -> `접수` -> `서빙` -> `배달 중` -> `배달 완료` -> `완료`
포장 주문: `대기` -> `접수` -> `서빙` -> `완료`
매장 주문: `대기` -> `접수` -> `서빙` -> `완료`

jordy-torvalds replied:
아 그렇군요! 
비슷한 부분이 있더라도 합치는 것 보단 별도로 분리하는게 유형별로 상태 플로우를 파악하기에 좋아서 전 나눠 놓는게 나은거 같다고 생각했었어요!
이전 정리 내용이 저는 더 이해가 잘되서 이전 상태로 돌리겠습니다!

hyunssooo commented on README.md:
```diff
@@ -96,8 +96,99 @@ docker compose -p kitchenpos up -d
+| 가격    | price         | 주문이 발생한 시점에 메뉴의 가격 |
+| 개수    | quantity      | 주문된 메뉴의 개수         |
+
+#### 주문 - 주문 테이블 용어사전
+
+| 한글명    | 영문명                     | 설명                                 |
```
주문 테이블도 주문의 하위 개념이라고 보신 걸까요? 
주문 테이블 또한, 메뉴 그룹과 마찬가지로 주문의 하위 개념은 아닌 것 같아서요.

jordy-torvalds replied:
생성과 변경 시점이 다르니 나눠지는게 맞겠네요! 피드백 반영하겟씁니다!

hyunssooo reviewed (approved):
안녕하세요. 
몇 가지 코멘트 남겼습니다.
다음 step 진행하시면서 확인 부탁드려요!
